ssl/statem/statem_clnt.c

```diff
@@ -1130,12 +1133,27 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
             }
         }
 
+        /* Fix session version if it is not reused and we are not in EAP-FAST */
+        if (s->session->tlsext_tick == NULL)
```

Comment:
> It doesn't look like RFC 4851 says anything about the PAC being somehow version-specific. I dunno, I suspect they didn't think about this very hard. Certainly the tls_session_secret_cb never fills an ssl_version value in.

Looks like you are right about it, fixed.

> (Or were you asking about why wpa_supplicant? This function was added for wpa_supplicant. They had some hacky patch for OpenSSL that eventually got upstreamed. I certainly hope no one else uses this code...)

Yeah, I was asking about it. So you say that whole EAP thing is someone's hack for OpenSSL? ;)